### PR TITLE
replaces InboundGridActorControl ad hoc actor lookup or start…

### PR DESCRIPTION
**Pull request by buritos**
…with the corresponding thread-safe methods in Stage.

### CODE REVIEW

**Comment by buritos** [comment]
requires: https://github.com/vlingo/vlingo-actors/pull/72

**Comment by buritos** [comment]
@VaughnVernon I will merge this so that the breaking changes are resolved. Please take a look at the changes to see if I have messed up something. Semantics should be the same except for `relocate` which previously had a bug. The bug was that in the scenario when there was no "thunk" running on the node, the code started a new actor instance that was not suspended.  Consequently, clients could deliver messages to the actor instance, before the relocation was complete.

File: src/test/java/io/vlingo/lattice/grid/Main.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public static void main(final String[] args) throws Exception {
       Grid grid = Grid.start("main-grid-world", Configuration.define(), ClusterProperties.allNodes(PORT_SEED), nodeName);
 
       System.out.println("WAITING.....");
-      Thread.sleep(30000);
+      Thread.sleep(10000);
```
Comment (VaughnVernon) [inline src/test/java/io/vlingo/lattice/grid/Main.java]: This is used only for test?

File: src/main/java/io/vlingo/actors/InboundGridActorControl.java [hunk 2]
```diff
@@ -72,12 +72,13 @@ public void forward(Id receiver, Id sender, Message message) {
   @Override
   public <T> void start(Id receiver, Id sender, Class<T> protocol, Address address, Definition.SerializationProxy definition) {
     logger.debug("Processing: Received application message: Start");
-    final GridActor<?> actor = (GridActor<?>) grid.actorAt(address);
-    if (actor == null) {
-      grid.actorFor(protocol, Definition.from(grid, definition, grid.world.defaultLogger()), address);
-    }
-    else if (actor.isSuspendedForRelocation()) {
-      logger.debug("Resuming thunk found at {} with definition='{}'", address, actor.definition());
+    final GridActor<?> actor = (GridActor<?>) grid.rawLookupOrStart(
```
Comment (VaughnVernon) [inline src/main/java/io/vlingo/actors/InboundGridActorControl.java]: At this point are there any possible exceptions?

File: src/main/java/io/vlingo/actors/InboundGridActorControl.java [hunk 3]
```diff
@@ -86,11 +87,9 @@ else if (actor.isSuspendedForRelocation()) {
   public <T> void deliver(
       Id receiver, Id sender, Returns<?> returns, Class<T> protocol, Address address, Definition.SerializationProxy definition, SerializableConsumer<T> consumer, String representation) {
     logger.debug("Processing: Received application message: Deliver");
-    Optional<Actor> maybeActor = Optional.ofNullable(grid.actorAt(address));
-    final Actor actor = maybeActor.orElseGet(() -> {
-      grid.actorThunkFor(protocol, Definition.from(grid, definition, grid.world.defaultLogger()), address);
-      return grid.actorAt(address);
-    });
+    final Actor actor = grid.actorLookupOrStartThunk(
```
Comment (VaughnVernon) [inline src/main/java/io/vlingo/actors/InboundGridActorControl.java]: Possible exceptions?

File: src/main/java/io/vlingo/actors/InboundGridActorControl.java [hunk 4]
```diff
@@ -99,12 +98,12 @@ else if (actor.isSuspendedForRelocation()) {
   @SuppressWarnings("unchecked")
   public void relocate(Id receiver, Id sender, Definition.SerializationProxy definition, Address address, Object snapshot, List<? extends io.vlingo.actors.Message> pending) {
     logger.debug("Processing: Received application message: Relocate");
-    final Optional<GridActor<?>> maybeActor = Optional.ofNullable((GridActor<?>) grid.actorAt(address));
-    final RelocationSnapshotConsumer<Object> consumer = maybeActor.map(a -> grid.actorAs(a, RelocationSnapshotConsumer.class))
-        .orElseGet(() -> grid.actorFor(
-            RelocationSnapshotConsumer.class, Definition.from(grid, definition, grid.world.defaultLogger()), address));
+    final GridActor<?> actor = (GridActor<?>) grid.actorLookupOrStartThunk(
```
Comment (VaughnVernon) [inline src/main/java/io/vlingo/actors/InboundGridActorControl.java]: Possible exceptions?

File: src/test/java/io/vlingo/lattice/grid/Main.java (hunk 1 above)
Comment (buritos) [reply src/test/java/io/vlingo/lattice/grid/Main.java]: yes

**Comment by buritos** [comment]
@VaughnVernon The new `ActorAddressAlreadyRegistered` is handled in those lookup methods. Other exceptions may occur, the same as in the previous version. Is there something in particular that you think should be handled?

**Comment by VaughnVernon** [comment]
@buritos Not necessarily. Only to make you aware in case it was overlooked.

**Pull request merged**